registry: add signature_directory_url parameter and fix data URI example (#92)
thibmeu/http-message-signatures-directory: draft-meunier-webbotauth-registry.md

  "rate-control": "429",
  "rate-expectation": "avg=10rps;max=100rps",
  "known-urls": ["/", "/robots.txt", "*.png"],

>   "signature_directory_url": "https://example.com/.well-known/http-message-signatures-directory",

Reviewer: the directory is a JWKS. I think it makes sense to have it there directly

```suggestion
  "jwks_uri": "https://example.com/.well-known/http-message-signatures-directory",
```

* `["/favicon.ico"]`
* `["/index.html"]`

> ## Signature Directory URL {#signature-agent-parameter-signature-directory-url}

Reviewer: ```suggestion
## JWKS URI {#signature-agent-parameter-jwks-uri}
```

> The `signature_directory_url` parameter provides the URL of the signature agent's

Reviewer: ```suggestion
The `jwks_uri` parameter provides the URL of the signature agent's
```

> HTTP Message Signatures Directory as defined in {{DIRECTORY}}.

Reviewer: ```suggestion
JWKS keys as defined in {{Section 5 of RFC7517}}. This covers
HTTP Message Signatures Directory as defined in {{DIRECTORY}}.
```

When present, this parameter separates key material discovery from metadata
discovery. Clients that need key material SHOULD fetch the directory at the
given URL rather than relying on the `keys` parameter. This separation allows
registry operators to host metadata and key material on different endpoints,
supporting deployment scenarios where the registry endpoint itself contains
signature agent card metadata but the key directory is hosted elsewhere.

If both `signature_directory_url` and `keys` are present, the

> `signature_directory_url` takes precedence for key discovery.

Reviewer: ```suggestion
If both `jwks_uri` and `keys` are present, the
`jwks_uri` takes precedence for key discovery.
```

Reviewer: this is just to merge, but I think we'll simply end up removing "keys" parameter

https://zerotrust-gateway.example.com/v1/signature-agent-card

# Below is an inlined card with the data URL scheme

> data:application/json,{"client_name":"Inline Bot","signature_directory_url":"https://inline.example.com/.well-known/http-message-signatures-directory"}

Reviewer: ```suggestion
data:application/json,{"client_name":"Inline Bot","jwks_uri":"https://inline.example.com/.well-known/http-message-signatures-directory"}
```

**Notes:**
: N/A

#### Signature Directory URL Parameter

**Parameter Name:**

> : signature_directory_url

Reviewer: ```suggestion
: jwks_uri
```

**Parameter Description:**

> : URL of the signature agent's HTTP Message Signatures Directory for key discovery

Reviewer: ```suggestion
: URL of the signature agent's key set. This can be an HTTP Message Signatures Directory for key discovery
```

# Changelog
{:numbered="false"}

v03

- Add optional `signature_directory_url` parameter to separate key material from metadata
- Fix inline data URL example in registry to use valid signature agent card

> 

Reviewer: we have not cut -02 yet. so this will be in -02
